### PR TITLE
As fix auth

### PR DESCRIPTION
**Pull request by artursmirnov**
Ticket: https://eu-app.assembla.com/spaces/Web/tickets/realtime_cardwall?ticket=222

This PR also includes some relevant clean up of the legacy code that has been refactored in scope of authorization.

The major update here is that the app now uses only one storage (either `localStorage` or `sessionStorage`) - the one that has been updated last, so no synchronization is involved anymore. Another important improvement is that the authorization information in the app now is read-only. The app only reads it and never writes. So it's always exactly the same that has been set by authorizer.

### CODE REVIEW

**Comment by artursmirnov** [comment]
It works for me locally and I expect it to work on test deployments, but lots of tests are failing. Nevertheless, it'd be good to get some early feedback if you have some time.

File: app/services/auth.js [hunk 1]
```diff
@@ -1,276 +1,196 @@
...
+  //           return Promise.reject();
+  //         }
+  //       })
+  //       .then(({ installation = null }) => {
+  //         this.currentUser.setProperties({ installation });
+  //       });
+  //   } else {
+  //     return Promise.resolve();
+  //   }
+  // },
```
Comment (rafael-travis) [inline app/services/auth.js]: Comment needed?

File: app/services/auth.js [hunk 2]
```diff
@@ -313,3 +220,18 @@ export default Service.extend({
...
+  const record = store.push(store.normalize('user', user));
+  const installation = store.peekAll('installation').findBy('owner.id', user.id) || null;
+  record.setProperties({ installation });
+  return record;
+}
+
+function runAfterSignOutCallbacks() {
+  afterSignOutCallbacks.forEach((callback) => {
+    callback();
+  });
```
Comment (rafael-travis) [inline app/services/auth.js]: would `afterSignOutCallbacks.forEach(callback)` work?

File: app/services/auth.js (hunk 1 above)
Comment (artursmirnov) [reply app/services/auth.js]: Yeah, this method is quite complex and I kept it for reference while fixing tests. I'll make sure to clean up before merge 👍

File: app/services/auth.js (hunk 2 above)
Comment (artursmirnov) [reply app/services/auth.js]: It should! Great suggestion, thanks 👍

Comment (artursmirnov) [reply app/services/auth.js]: Ah, no, it won't. `callback` is a function that's passed to iterator, so it's undefined outside of it.

Comment (artursmirnov) [reply app/services/auth.js]: Though, I did simplify it like this: `afterSignOutCallbacks.forEach(callback => callback());`

Comment (rafael-travis) [reply app/services/auth.js]: Ah, right lol, I misread it. Nice!

**Comment by artursmirnov** [comment]
Hey folks, this PR is green now and ready for the final review 🎉 Could you please take a look when you have time?

File: app/models/user.js [hunk 3]
```diff
@@ -1,45 +1,37 @@
...
   allowMigration: attr('boolean'),
 
-
   type: 'user',
 
-  fullName: computed('name', 'login', function () {
-    let name = this.name;
-    let login = this.login;
-    return name || login;
+  fullName: or('name', 'login'),
```
Comment (piotr-travisci) [inline app/models/user.js]: Like Scouts do it:
"Leave it better than you found it."
👏

**Pull request merged**